osdep/io.c

```diff
@@ -484,6 +484,20 @@ int mp_creat(const char *filename, int mode)
     return mp_open(filename, _O_CREAT | _O_WRONLY | _O_TRUNC, mode);
 }
 
+int mp_rename(const char *oldpath, const char *newpath)
```

Comment:
It is fine, but name it `replace`, it has completely different semantic than POSIX's `rename`. See https://docs.python.org/3/library/os.html#os.replace (and rename) how they handle this difference.

> If you want cross-platform overwriting of the destination, use replace()

rename on windows fails when destination exists.

Reply:
POSIX `rename` allows replacing so where is the difference?

Reply:
Windows `rename` does not.

Reply:
I know. The goal of `mp_rename` is to act like the POSIX function and it does just that.
Very much in line with the rest of the osdep wrappers so I don't see why it should have a different name.